homeassistant/components/mqtt/cover.py

```diff
@@ -625,7 +625,7 @@ async def async_set_cover_tilt_position(self, **kwargs):
                 "tilt_min": self._config[CONF_TILT_MIN],
                 "tilt_max": self._config[CONF_TILT_MAX],
             }
-            tilt = template.async_render(parse_result=False, variables=variables)
+            tilt = MqttCommandTemplate(template).async_render(None, variables=variables)
```

Comment:
This seems a bit strange, we instantiate a new template every time `async_set_cover_tilt_position` is called? And how does the template work without a `hass` reference?

Reply:
The template it self holds the hass reference. It was assigned at line 297.

Reply:
Unified the code for cover and alarm_control_panel. MqttCommandTemplate is now set up at _setup_from_config.